### PR TITLE
Add Get IPV4/6 from Network service

### PR DESCRIPTION
**Pull request by Lanpingner**
This PR simply add IP info that can be collected from the Network service both from WiFi or Wired

### How to Implement to your code

```
const { wifi } = await Service.import("network")
Widget.Box({
class_name: "ipcontainer",
expand: true,
children: [
    Widget.Label({
        class_name: "iplabel",
        label: network.wifi.nmDevice.bind("ipv4_address").as(ipv4_address => ipv4_address || "No wifi IP")
    }),
    Widget.Label({ label: " " }),
    Widget.Label({
        class_name: "iplabel",
        label: network.wired.nmDevice.bind("ipv4_address").as(ipv4_address => ipv4_address || "No wired IP"),

    }),
]}),
```

### CODE REVIEW

**Comment by Aylur** [comment]
they should be defined as gobject properties and they should connect to the ip config object to listen for changes

**Comment by Lanpingner** [comment]
I am not sure if this is what you meant, sorry very new to Typescript, i am open for suggestions.
If this is not what you meant, could you please provide me with an example or a docs

**Comment by Aylur** [comment]
is it possible for the ip4 and ip6 config objects to change on the device? should be handled if they can
the network service really needs some work, maybe we should make a base class for Wired and Wifi, but thats for another PR

**Comment by topaxi** [comment]
Came here to request this feature or potentially contribute, glad to see this already being pursued :+1:

File: src/service/network.ts [hunk 1]
```diff
@@ -221,6 +247,8 @@ export class Network extends Service {
             'wired': ['jsobject'],
             'primary': ['string'],
             'connectivity': ['string'],
+            'ipv4-address': ['string'],
+            'ipv6-address': ['string'],
```
Comment (Aylur) [inline src/service/network.ts:251]: why is this here?

File: src/service/network.ts [hunk 2]
```diff
@@ -131,6 +142,8 @@ export class Wifi extends Service {
 
     get enabled() { return this._client.wireless_enabled; }
     set enabled(v) { this._client.wireless_enabled = v; }
+    get ipv4Address() { return this._device.ip4_config.get_addresses(); }
+    get ipv6Address() { return this._device.ip6_config.get_addresses(); }
```
Comment (Aylur) [inline src/service/network.ts:146]: the getter needs to be in snake_case for the gobject getter to work

File: src/service/network.ts [hunk 3]
```diff
@@ -89,6 +92,14 @@ export class Wifi extends Service {
...
             this._activeAp();
         }
+        const ip4Config = this._device.ip4_config;
+        const ip6Config = this._device.ip6_config;
+
+        if ((this._device) && (ip4Config))
+            ip4Config.connect('notify::addresses', () => this.changed('ipv4-address'));
+
+        if ((this._device) && (ip6Config))
+            ip6Config.connect('notify::addresses', () => this.changed('ipv6-address'));
```
Comment (Aylur) [inline src/service/network.ts:102]: I don't know much about networks, but is the ip6_config object on the device permanent? becaue if its not this would need to be disconnected when it changes

**Review by Aylur — COMMENTED** [review]
We should have a Device subclass, to avoid code repetition
```
Device extends Service
Wifi extends Device
Wired extends Device
```

File: src/service/network.ts (hunk 1 above)
Comment (Lanpingner) [reply src/service/network.ts:251]: Removed in the next commit, it was just an idea to make it one place where you can collect info, by accident i pushed this

File: src/service/network.ts (hunk 2 above)
Comment (Lanpingner) [reply src/service/network.ts]: fixed in the commit

**Comment by Lanpingner** [comment]
> We should have a Device subclass, to avoid code repetition
> 
> ```
> Device extends Service
> Wifi extends Device
> Wired extends Device
> ```
I will see what i can come up with
